docs/_includes/anchor_links.html

```diff
@@ -1,33 +1,38 @@
 <script>
-  var anchorForId = function (id) {
-    var anchor = document.createElement("a");
-    anchor.className = "header-link";
-    anchor.href      = "#" + id;
-    anchor.innerHTML = "<span class=\"sr-only\">Permalink</span><i class=\"fa fa-link\"></i>";
-    anchor.title = "Permalink";
-    return anchor;
-  };
+/* Creates an anchor element with the given ID and link for the permalink*/
+const anchorForId = (id) => {
+  const anchor = document.createElement("a");
+  anchor.className = "header-link";
+  anchor.href = `#${id}`;
+  anchor.innerHTML = `<span class="sr-only">Permalink</span><i class="fa fa-link" aria-hidden="true"></i>`;
+  anchor.title = "Permalink";
+  return anchor;
+};
 
-  var linkifyAnchors = function (level, containingElement) {
-    var headers = containingElement.getElementsByTagName("h" + level);
-    for (var h = 0; h < headers.length; h++) {
-      var header = headers[h];
-
-      if (typeof header.id !== "undefined" && header.id !== "") {
-        header.appendChild(anchorForId(header.id));
-      }
+/* Finds all headers of the specified level within the given element, and adds a permalink to each header*/
+const linkifyAnchors = (level, containingElement) => {
+  /* Retrieve all headers with the specified level as an Array*/
+  const headers = Array.from(containingElement.getElementsByTagName(`h${level}`));
+  /* For each header with an ID, add a permalink*/
+  headers.forEach((header) => {
+    if (header.id) {
+      header.appendChild(anchorForId(header.id));
```

Comment:
The comment at the top of the method is great. The comments that duplicate the code in the function aren't as great because the code is straightforward enough to read it and know what it's doing without the comment. What do you think about letting the code stand alone where it's completely obvious?
```suggestion
/* Finds all headers of the specified level within the given element, and adds a permalink to each header*/
const linkifyAnchors = (level, containingElement) => {
  const headers = Array.from(containingElement.getElementsByTagName(`h${level}`));
  headers.forEach((header) => {
    if (header.id) {
      header.appendChild(anchorForId(header.id));
```

Reply:
Hey @mattr-, agreed. I've added comments for an easier understanding of the script's purpose. Since I wasn't certain who on the team would review it, I'd typically suggest removing these comments before merging and figured we might be using some sort of compressor that removes these comments anyway.

Following your suggestion.